File: core/src/test/java/javabot/web/views/AdminViewTest.java
package javabot.web.views;

import io.dropwizard.views.freemarker.FreemarkerViewRenderer;
import net.htmlparser.jericho.Source;

import javax.servlet.http.HttpServletRequest;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.Locale;

public class AdminViewTest extends ViewsTest {
    protected HttpServletRequest getRequest() {
        return new MockServletRequest(true);
    }


    protected Source render() throws IOException {
        FreemarkerViewRenderer renderer = new FreemarkerViewRenderer();
        ByteArrayOutputStream output = new ByteArrayOutputStream();
        renderer.render(new AdminIndexView(getInjector(), getRequest()), Locale.getDefault(), output);
        return new Source(new ByteArrayInputStream(output.toByteArray()));
    }

}

File: src/test/kotlin/javabot/web/views/AdminViewTest.kt
package javabot.web.views

import jakarta.servlet.http.HttpServletRequest
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import javabot.model.Admin
import net.htmlparser.jericho.Source

open class AdminViewTest : ViewsTest() {
    protected fun getRequest(): HttpServletRequest {
        return MockServletRequest(true)
    }

    protected fun render(): Source {
        val output = ByteArrayOutputStream()
        val templateInstance =
            templateService.createAdminIndexView(getRequest(), adminDao.findAll()[0], Admin())
        val html = templateInstance.render()
        output.write(html.toByteArray())
        return Source(ByteArrayInputStream(output.toByteArray()))
    }
}
